app/uk/gov/hmrc/lisaapi/controllers/AccountController.scala

```diff
@@ -38,357 +38,349 @@ class AccountController extends LisaController with LisaConstants {
-            auditService.audit(
-              auditType = "accountTransferred",
-              path = getCreateOrTransferEndpointUrl(lisaManager),
-              auditData = transferRequest.toStringMap + (ZREF -> lisaManager)
-            )
-
-            val data = ApiResponseData(message = "Account transferred", accountId = Some(accountId))
-
-            LisaMetrics.incrementMetrics(startTime, CREATED, LisaMetricKeys.ACCOUNT)
-
-            Created(Json.toJson(ApiResponse(data = Some(data), success = true, status = CREATED)))
-          case CreateLisaAccountInvestorNotFoundResponse =>
-            handleCreateOrTransferFailure(lisaManager, transferRequest, ErrorInvestorNotFound, FORBIDDEN, action)
-          case CreateLisaAccountInvestorComplianceCheckFailedResponse =>
-            handleCreateOrTransferFailure(lisaManager, transferRequest, ErrorInvestorComplianceCheckFailedCreateTransfer, FORBIDDEN, action)
-          case CreateLisaAccountInvestorPreviousAccountDoesNotExistResponse =>
-            handleCreateOrTransferFailure(lisaManager, transferRequest, ErrorPreviousAccountDoesNotExist, FORBIDDEN, action)
-          case CreateLisaAccountInvestorAccountAlreadyClosedResponse =>
-            handleCreateOrTransferFailure(lisaManager, transferRequest, ErrorAccountAlreadyClosed, FORBIDDEN, action)
-          case CreateLisaAccountInvestorAccountAlreadyVoidResponse =>
-            handleCreateOrTransferFailure(lisaManager, transferRequest, ErrorAccountAlreadyVoided, FORBIDDEN, action)
-          case CreateLisaAccountAlreadyExistsResponse =>
-            handleCreateOrTransferFailure(lisaManager, transferRequest, ErrorAccountAlreadyExists(transferRequest.accountId), CONFLICT, action)
-          case CreateLisaAccountErrorResponse =>
-            handleCreateOrTransferFailure(lisaManager, transferRequest, ErrorInternalServerError, INTERNAL_SERVER_ERROR, action)
-        } recover {
-          case e: Exception =>
-            Logger.error(s"AccountController: An error occurred in due to ${e.getMessage} returning internal server error")
-            handleCreateOrTransferFailure(lisaManager, transferRequest, ErrorInternalServerError, INTERNAL_SERVER_ERROR, action)
-        }
+  def getAccountDetails(lisaManager: String, accountId: String): Action[AnyContent] =
+    (validateHeader() andThen validateLMRN(lisaManager) andThen validateAccountId(accountId)).async { implicit request =>
+      implicit val startTime: Long = System.currentTimeMillis()
+      withEnrolment(lisaManager) { (_) =>
+        processGetAccountDetails(lisaManager, accountId)
       }
     }
 
-    private def hasValidDatesForCreation(lisaManager: String, creationRequest: CreateLisaAccountCreationRequest)
-                                        (success: () => Future[Result])
-                                        (implicit hc: HeaderCarrier, startTime:Long): Future[Result] = {
+  private def processGetAccountDetails(lisaManager: String, accountId: String)
```

Comment:
Given how much shorter the `getAccountDetails` function has become, I feel like perhaps `processGetAccountDetails` should just be inlined now?

Reply:
Agreed.

Reply:
I've made this change now.